CDCgov/prime-reportstream
# 7052 - UP shows routing filter results in submission history

These changes continue on the work from #7051 to ensure that the routing filter results show up in the submission history. There was a slight bug that was preventing this from working, but once that was resolved, this worked basically out-of-the-box.

Test Steps:
1. Add a routing filter to `organizations.yml`
2. Reload settings and run ReportStream
3. Send a report that will fail to pass the routing filter
4. Using the report id from the result of the report submission, query the submission history

<details>
<summary>You should see a result like this</summary>

```
{
    "id": "884c19ae-1499-4313-9ed5-eae8e604756e",
    "submissionId": 37,
    "overallStatus": "Not Delivering",
    "timestamp": "2022-12-14T02:11:29.136Z",
    "plannedCompletionAt": null,
    "actualCompletionAt": null,
    "sender": "ignore.ignore-full-elr",
    "reportItemCount": 1,
    "errorCount": 0,
    "warningCount": 0,
    "httpStatus": 201,
    "destinations": [
        {
            "organization": "FOR TESTING ONLY",
            "organization_id": "ignore",
            "service": "FULL_ELR",
            "itemCount": 0,
            "itemCountBeforeQualityFiltering": 1,
            "filteredReportRows": [
                "For ignore.FULL_ELR, filter [matches(test_result_status, C, P)][] filtered out item 1234d1d1-95fe-462c-8ac6-46728dba581c"
            ],
            "filteredReportItems": [
                {
                    "filterType": "ROUTING_FILTER",
                    "filterName": "[matches(test_result_status, C, P)]",
                    "filteredTrackingElement": "1234d1d1-95fe-462c-8ac6-46728dba581c",
                    "filterArgs": [],
                    "message": "For ignore.FULL_ELR, filter [matches(test_result_status, C, P)][] filtered out item 1234d1d1-95fe-462c-8ac6-46728dba581c"
                }
            ],
            "sentReports": [],
            "downloadedReports": []
        }
    ],
    "actionName": "receive",
    "externalName": null,
    "reportId": "884c19ae-1499-4313-9ed5-eae8e604756e",
    "topic": "full-elr",
    "errors": [],
    "warnings": [],
    "destinationCount": 0
}
```

</details>


## Changes
- Every time I edit the `organizations.yml` file, my IDE fixes the formatting, and it makes it hard to keep track of what I changed, so I'm including **format-only changes** in that file to prevent that from continuing to happen in the future.
- In `FHIRRouter.kt`, we were logging filter results multiple times since a non-passing filter at one step would continue to hit all of the other log points meant for the other filter types. This also meant that we were logging non-passing juris filters, and that we often didn't even have all of the variables set we need for the log messages on other filters (including routing). I added some continues for the `!passes` cases for each filter, which fixes this issue, and enhances readability. Victor also added a helper function which consolidates some duplicated code.
- Victor added several tests, including ones which would have caught this issue

## Checklist

### Testing

## Linked Issues
- Fixes #7052
